engine/src/flutter/shell/platform/darwin/macos/framework/Source/FlutterRunLoop.mm

```diff
@@ -0,0 +1,130 @@
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterRunLoop.h"
+#include "fml/logging.h"
+
+#include <vector>
+
+namespace {
+struct Task {
+  void (^block)(void);
+  CFAbsoluteTime target_time;
+};
+
+CFStringRef kFlutterRunLoopMode = CFSTR("FlutterRunLoopMode");
+
+FlutterRunLoop* mainLoop;
+
+}  // namespace
+
+@implementation FlutterRunLoop (Private)
+
+CFRunLoopRef _runLoop;
+CFRunLoopSourceRef _source;
+CFRunLoopTimerRef _timer;
+std::vector<Task> _tasks;
+
```

Comment:
Shouldn't these be in an ivar block? I'm actually kind of surprised that this compiles; I wonder if they're being treated as globals?

```suggestion
@implementation FlutterRunLoop (Private) {
  CFRunLoopRef _runLoop;
  CFRunLoopSourceRef _source;
  CFRunLoopTimerRef _timer;
  std::vector<Task> _tasks;
}
```

Reply:
Ouch.